src/pyedb/dotnet/edb_core/components.py

```diff
@@ -872,22 +875,21 @@ def create_port_on_pins(
             pins = cmp_pins
         if not len([pin for pin in pins if isinstance(pin, EDBPadstackInstance)]) == len(pins):
             self._logger.error("Pin list must contain only pins instances")
-            return
+            return False
         if not port_name:
             port_name = "Port_{}_{}".format(pins[0].net_name, pins[0].name)
         if len([pin for pin in reference_pins if isinstance(pin, str)]) == len(reference_pins):
             ref_cmp_pins = []
             for ref_pin_name in reference_pins:
                 if ref_pin_name in refdes_pins:
                     ref_cmp_pins.append(refdes_pins[ref_pin_name])
-                elif "-" in ref_pin_name and ref_pin_name.split("-")[1] in refdes_pins:
-                    ref_cmp_pins.append(refdes_pins[ref_pin_name.split("-")[1]])
+                elif "-" in ref_pin_name:
+                    if ref_pin_name.split("-")[1] in refdes_pins:
+                        ref_cmp_pins.append(refdes_pins[ref_pin_name.split("-")[1]])
             if not ref_cmp_pins:
```

Comment:
Should it raise Error instead of error in log. I am afraid no one will check the log unless the code breaks.

Reply:
Depends on what side you want to side on. Either providing something "resilient" or that breaks when it fails.
As a scientist (and project maintainer) I'd go with the second approach.
I prefer users (or myself) being notified early that something wrong is going on than deducing wrong stuff because I didn't notice a wrong behavior in the process.

Reply:
Like we spoke lrt's keep it like this for now but I understand your point.